rclcpp_lifecycle/test/test_lifecycle_node.cpp

```diff
@@ -492,31 +492,51 @@ TEST_F(TestDefaultStateMachine, test_getters) {
 TEST_F(TestDefaultStateMachine, test_graph) {
   auto test_node = std::make_shared<EmptyLifecycleNode>("testnode");
   auto names = test_node->get_node_names();
-  EXPECT_EQ(names.size(), 1u);
-  EXPECT_STREQ(names[0].c_str(), "/testnode");
+
+  EXPECT_NE(names.end(), std::find(names.begin(), names.end(), std::string("/testnode")));
 
   // parameter_events, rosout, /testnode/transition_event
   auto topic_names_and_types = test_node->get_topic_names_and_types();
-  EXPECT_EQ(topic_names_and_types.size(), 3u);
+  EXPECT_NE(
+    topic_names_and_types.end(),
+    topic_names_and_types.find(std::string("/testnode/transition_event")));
   EXPECT_STREQ(
     topic_names_and_types["/testnode/transition_event"][0].c_str(),
     "lifecycle_msgs/msg/TransitionEvent");
```

Comment:
Two separate things here:

1.  This test isn't *quite* the same as what was there before.  In particular, I think we need to ensure that 3 different topics exist here: `/testnode/transition_event`, `/testnode/rosout`, and `/testnode/parameter_events`.
1.  The test for whether `/testnode/transition_event` exists needs to be an ASSERT, otherwise we'll crash when indexing into the array in the second part of the test.

Reply:
Opened discussion in private about point 1. I agree with point 2, changed in newest commit.

Reply:
I think the conclusion of our discussion was that it is fine to remove the checks for the other topics.  But in that case, please update the comment on line 498 to make it explicit what we are testing.

Reply:
Updated comment.